pkg/grpc/server_test.go

```diff
@@ -91,6 +96,201 @@ func (a *APIServerSuite) Test_CustomAPI() {
 	})
 }
 
+func (a *APIServerSuite) Test_Server_RateLimit_HTTP_Integration() {
```

Comment:
If you extract following function then it could be used in all tests 
```go
func (a *APIServerSuite) runTest(rateLimit int) {
	hitLimit := false
	for i := 0; i < 50; i++ {
		resp, err := http.Get(testUrl)
		a.Require().NoError(err)

		if i < rateLimit {
			a.Require().Equal(http.StatusOK, resp.StatusCode)
			continue
		}

		if resp.StatusCode != 200 {
			a.Require().Equal(http.StatusTooManyRequests, resp.StatusCode)
			hitLimit = true
			break
		}
	}
	if rateLimit > 0 {
		a.Assert().True(hitLimit)
	}

	// Wait for rate limit to refill.
	time.Sleep(2 * time.Second)

	resp, err := http.Get(testUrl)
	a.Assert().NoError(err)
	a.Assert().Equal(http.StatusOK, resp.StatusCode)
}
```
grpc tests also tests number of requests. Should we have same options in HTTP?